packages/core/src/components/StepList/StepList.scss

```diff
@@ -0,0 +1,356 @@
+@import '@cmsgov/design-system-support/src/settings/index';
+
+/*
+Step List
+
+@status beta
+
+A step list represents a user's progression through an application or multi-page
+form. It serves as a table of contents and a way to quickly see where they are
+and what they should be working on next.
+
+Markup:
+<div style="max-width: 628px">
+  <ol class="ds-c-step-list ds-u-margin-top--4">
+    <li class="ds-c-step ds-c-step--completed">
+      <h2 class="ds-c-step__header">
+        <div class="ds-c-step__title">Choose a tax year</div>
+        <div class="ds-c-step__actions">
```

Comment:
Depending on the conclusion of [the discussion around the placement of these actions on mobile and screen readers](https://teamwds.slack.com/archives/C7S6GJKC2/p1512149247000013), we may need to adjust the markup of this section. Based on the current discussion, it sounds like we want these actions read after the title and description, so this markup would need moved after those. I think this actually makes life easier for us when it comes to how this adjusts for mobile and desktop. We could change the markup so that it's two columns, which collapse to one column for small screens. Here's some pseudo code of what I had in mind when working on the design of this...

```html
<li class="ds-c-step">
  <!-- Left column on desktop -->
  <div class="ds-c-step__body">
    <h2 class="ds-c-step__header">Foo</h2>
    <p>Description text</p>
    <ol class="ds-c-step__substeps">...</ol>
  </div>

  <!-- Right column on desktop -->
  <div class="ds-c-step__actions">
     <a class="ds-c-button" href="#">Start</a>
  </div>
</li>
```

The only tricky thing this introduces is the alignment of the substep "Edit" links, but I think this is easier to solve with absolute positioning (`right: 0`).

Reply:
I pulled Patrick's code base, and mocked up some HTML by hacking on the example view. This moves the primary actions to the bottom of a `<Step />`, beneath the description and the sub-actions. It also adds header 3's for the description, sub-actions, and actions, that are available only to screen readers. These could easily be updated via className to be visible if wanted.

```
<div style="max-width: 628px">
  <ol class="ds-c-step-list ds-u-margin-top--4">
    <li class="ds-c-step ds-c-step--completed">
      <h2 class="ds-c-step__header">
        <span class="ds-c-step__title">Choose a tax year</span>
      </h2>
      <div class="ds-c-step__actions" aria-labelledby="step-1-actions">
        <h3 class="ds-u-visibility--screen-reader" id="step-1-actions">Primary actions for Choose a Tax Year</h3>
        <div class="ds-c-step__completed-text">
          <span class="ds-u-visibility--screen-reader">"Choose a tax year"</span> Completed
        </div>
        <a href="#step-1">
          Edit<span class="ds-u-visibility--screen-reader">
            {" "}
            "Choose a tax year"
          </span>
        </a>
      </div>
    </li>
    <li class="ds-c-step ds-c-step--current">
      <h2 class="ds-c-step__header">
        <span class="ds-c-step__title">Enter household details</span>
      </h2>
      <div aria-labelledby="step-2-desc" class="ds-c-step__description">
        <h3 class="ds-u-visibility--screen-reader" id="step-2-desc">
          Step 2 Description
        </h3>
        <p>
          Answer questions about who in your household qualifies for a premium tax
          credit and information on each person, including date of birth,
          location(s) they lived in for the year, and months of marketplace
          coverage.
        </p>
      </div>
      <div aria-describedby="step-2-sub-actions" class="ds-c-step__substep">
        <h3 class="ds-u-visibility--screen-reader" id="step-2-sub-actions" >Secondary actions for Household Details</h3>
        <ol class="ds-c-step__substeps">
          <li class="ds-c-substep">
            <div class="ds-c-substep__line">
              <div class="ds-c-substep__title">Overall household</div>
              <a href="#step-2a">
                Edit<span class="ds-u-visibility--screen-reader">
                  {" "}
                  "Overall household"
                </span>
              </a>
            </div>
          </li>
          <li class="ds-c-substep">
            <div class="ds-c-substep__line">
              <div class="ds-c-substep__title">Bob's information</div>
            </div>
          </li>
          <li class="ds-c-substep">
            <div class="ds-c-substep__line">
              <div class="ds-c-substep__title">Barb's information</div>
            </div>
          </li>
        </ol>
      </div>
      <div aria-labelledby="step-2-actions" class="ds-c-step__actions">
        <h3 class="ds-u-visibility--screen-reader" id="step-2-actions">Primary actions for Household Details</h3>
        <a href="#step-2" class="ds-c-button ds-c-button--primary">
          Resume<span class="ds-u-visibility--screen-reader">
            {" "}
            "Enter household details"
          </span>
        </a>
      </div>
    </li>
    <li class="ds-c-step">
      <h2 class="ds-c-step__header">
        <div class="ds-c-step__title">Review your information</div>
        <div class="ds-c-step__actions">
      </div></h2>
    </li>
    <li class="ds-c-step">
      <h2 class="ds-c-step__header">
        <div class="ds-c-step__title">View premium results</div>
        <div class="ds-c-step__actions">
      </div></h2>
    </li>
  </ol>
</div>
```

Reply:
@1Copenut I'm working on implementing your headings and described-bys and such. Do you have examples of sub-action headings and description headings that reuse the step titles? Should we be saying "Description for "Enter household details""? I haven't been passing in the step number, but I can do that if you'd prefer it to be "Step 2 Description"